test/converters/converters.go

```diff
@@ -0,0 +1,21 @@
+package converters
+
+import (
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters/junitxml"
+	"github.com/bitrise-io/steps-deploy-to-bitrise-io/test/converters/xcresult"
+)
+
+type Intf interface {
```

Comment:
Supposedly this step is the main user of this interface. As long as there are 2 types (android, ios) this interface feels a bit complex for the intended use. It all depends how it is used in the client code, but it could be a function call instead of a for with 3 function calls. Let's discuss.

Reply:
4+ converters are about to be implemented

Reply:
Ok, what about 1 function then, XML([]string) ([]byte, error)

Reply:
we need to separate the logic for finding supported converter(s) and if they support the current test result type then we need to be able to run them

Reply:
The interface methods reduced to XML() and Detect(). I think now both of these describe the converter properly. XML() is a Go convention for typed outputs like String() etc..